src/history/FileTransferInfo.h

```diff
@@ -36,21 +35,44 @@ class FileTransferInfo
     }
 
     FileTransferInfo(TmpDir const& snapDir, std::string const& snapType,
-                     uint32_t checkpointLedger)
+                     uint32_t checkpointLedger, bool cascade = false)
         : mType(snapType)
         , mHexDigits(fs::hexStr(checkpointLedger))
-        , mLocalPath(snapDir.getName() + "/" + baseName_nogz())
+        , mLocalPath(getLocalDir(snapDir, snapType, cascade) + "/" +
+                     baseName_nogz())
     {
     }
 
     FileTransferInfo(TmpDir const& snapDir, std::string const& snapType,
-                     std::string const& hexDigits)
+                     std::string const& hexDigits, bool cascade = false)
         : mType(snapType)
         , mHexDigits(hexDigits)
-        , mLocalPath(snapDir.getName() + "/" + baseName_nogz())
+        , mLocalPath(getLocalDir(snapDir, snapType, cascade) + "/" +
```

Comment:
why is `getLocalDir` taking `snapType` as a parameter but not `hexDigits` ? I imagine it should take neither as they are member variables

Reply:
good catch